qiskit/providers/ibmq/job/ibmqjob.py

```diff
@@ -160,23 +160,15 @@ def qobj(self) -> Qobj:
     def properties(self) -> Optional[BackendProperties]:
         """Return the backend properties for this job.
 
-        The properties might not be available if the job hasn't completed,
-        in which case None is returned.
-
         Returns:
-            BackendProperties: the backend properties used for this job, or None if
-                properties are not available.
+            BackendProperties: the backend properties used for this job.
 
         Raises:
             JobError: if there was some unexpected failure in the server.
         """
         with api_to_job_error():
             properties = self._api.job_properties(job_id=self.job_id())
 
-        # Backend properties of a job might not be available if the job hasn't
-        # completed. This is to ensure the properties returned are up to date.
-        if not properties:
```

Comment:
Can you double-check if we can really remove this check? In particular:
* for jobs against a simulator, properties is probably empty
* for older jobs, can we be sure that 100% properties will be available (ie. edge cases such as errored jobs or some odd combinations)

Reply:
You were right. Forgot about the simulator. Put the check back.